test/common/common/thread_test.cc

```diff
@@ -190,6 +203,41 @@ TEST_F(ThreadAsyncPtrTest, ManagedAlloc) {
   }
 }
 
+TEST_F(ThreadAsyncPtrTest, TruncateWait) {
+  absl::Notification notify;
+  auto thread = thread_factory_.createThread([&notify]() { notify.WaitForNotification(); },
+                                             Options{"this name is way too long for posix"});
+  notify.Notify();
+
+  // To make this test work on multiple platforms, just assume the first 10 characters
+  // are retained.
+  EXPECT_THAT(thread->name(), testing::StartsWith("this name "));
+  thread->join();
```

Comment:
note the calls to join() here. Do we think we should auto-join on destruct? Without these there are tsan races.

Reply:
Either auto-join on destruct, or ASSERT that the thread is already joined in destruct.

Reply:
After brief discussion we'll assert on destruct, as join() may block and that may not be expected.